### PR TITLE
Allow user to unlisten to queries after shutdown.

### PR DESCRIPTION
**Pull request by wu-hui**

### CODE REVIEW

File: packages/firestore/src/core/firestore_client.ts [hunk 1]
```diff
@@ -536,7 +536,6 @@ export class FirestoreClient {
   }
 
   unlisten(listener: QueryListener): void {
-    this.verifyNotShutdown();
     this.asyncQueue.enqueueAndForget(() => {
```
Comment (schmidt-sebastian) [inline packages/firestore/src/core/firestore_client.ts]: This potentially calls into the SyncEngine. I wonder if we should guard this on the shutdown state and not call `unlisten` if we are shut down?

Comment (wu-hui) [reply packages/firestore/src/core/firestore_client.ts]: Done.

File: packages/firestore/src/core/firestore_client.ts [hunk 2]
```diff
@@ -536,7 +536,10 @@ export class FirestoreClient {
   }
 
   unlisten(listener: QueryListener): void {
-    this.verifyNotShutdown();
+    // Make unlisten a no-op if client is shut down.
```
Comment (schmidt-sebastian) [inline packages/firestore/src/core/firestore_client.ts]: Your code is so clear that I think we can remove this comment :)

File: packages/firestore/src/core/firestore_client.ts [hunk 3]
```diff
@@ -536,7 +536,10 @@ export class FirestoreClient {
...
+    if(this.clientShutdown) {
```
Comment (schmidt-sebastian) [inline packages/firestore/src/core/firestore_client.ts]: There is something going on with your formatter. This should be `if (this.clientShutdown)`

I would also slightly prefer if you wrote the inverse and removed the early return. Optional.

File: packages/firestore/src/core/firestore_client.ts (hunk 2 above)
Comment (wu-hui) [reply packages/firestore/src/core/firestore_client.ts]: I actually want to keep it. The reason is `FirestoreClient` methods generally run `verifyNotShutdown` before doing anything. This one deviate from that. 

I updated the comment to be more clear.

File: packages/firestore/src/core/firestore_client.ts (hunk 3 above)
Comment (wu-hui) [reply packages/firestore/src/core/firestore_client.ts]: Yeah, it's strange, the automated prettier commit did not make into the push somehow..

**Pull request merged**